sunpy/net/jsoc/jsoc.py

```diff
@@ -765,11 +766,17 @@ def _lookup_records(self, iargs):
             key = keywords
         log.debug(f"Running following query: {ds}")
         log.debug(f"Requesting following keywords: {key}")
-        result = client.query(ds, key=key, rec_index=isMeta)
-        if result is None or result.empty:
+        result = client.query(ds, key=key, seg=segments_passed, rec_index=isMeta)
+        # DRMS returns a tuple of length 2 if there is segment information
+        if len(result) == 2:
```

Comment:
This is not the cleanest code. Needs some improvment.

Reply:
Could you instead check explicitly segments are being passed?